crates/ruff_python_semantic/src/context.rs

```diff
@@ -113,6 +114,55 @@ impl<'a> Context<'a> {
             .map_or(false, |binding| binding.kind.is_builtin())
     }
 
+    /// Resolve a reference to the given symbol.
+    pub fn resolve_reference(&mut self, symbol: &str, range: TextRange) -> ResolvedReference {
+        let mut first_iter = true;
+        let mut import_starred = false;
+        for scope_id in self.scopes.ancestor_ids(self.scope_id) {
+            let scope = &self.scopes[scope_id];
+            if scope.kind.is_class() {
+                if symbol == "__class__" {
+                    return ResolvedReference::ImplicitGlobal;
+                } else if !first_iter {
```

Comment:
i'd use `self.scopes.ancestor_ids(self.scope_id).enumerate()` instead, this also avoid `continue` and `first_iter` having bad interactions